sdk/core/azure_core_amqp/src/fe2o3/management.rs

```diff
@@ -67,6 +71,18 @@ impl AmqpManagementApis for Fe2o3AmqpManagement {
         })?;
         Ok(())
     }
+
+    async fn detach(mut self) -> Result<()> {
+        // Detach the management client from the session.
+        let management = self
+            .management
+            .take()
+            .ok_or_else(|| Error::message(ErrorKind::Other, "Unattached management node."))?;
```

Comment:
Tip: since you repeat this pattern a lot, you could actually define a `fn` and just pass it in along with your str. No biggie, though.

Reply:
That's not a bad idea at all. I'm going to be revisiting the whole error strategy soon, I'll keep this in mind for that.